include/experimental/__p2630_bits/submdspan_mapping.hpp

```diff
@@ -120,85 +154,97 @@ struct preserve_layout_left_mapping<std::index_sequence<Idx...>, SubRank,
 // and the diagnostic push. I tried really hard to find something shorter
 // but no luck ...
 #if defined __NVCC__
-    #ifdef __NVCC_DIAG_PRAGMA_SUPPORT__
-        #pragma nv_diagnostic push
-        #pragma nv_diag_suppress = implicit_return_from_non_void_function
-    #else
-      #ifdef __CUDA_ARCH__
-        #pragma diagnostic push
-        #pragma diag_suppress implicit_return_from_non_void_function
-      #endif
-    #endif
+#ifdef __NVCC_DIAG_PRAGMA_SUPPORT__
+#pragma nv_diagnostic push
+#pragma nv_diag_suppress = implicit_return_from_non_void_function
+#else
+#ifdef __CUDA_ARCH__
+#pragma diagnostic push
+#pragma diag_suppress implicit_return_from_non_void_function
+#endif
+#endif
 #elif defined __NVCOMPILER
-    #pragma    diagnostic push
-    #pragma    diag_suppress = implicit_return_from_non_void_function
+#pragma diagnostic push
+#pragma diag_suppress = implicit_return_from_non_void_function
 #endif
 // Actual submdspan mapping call
 template <class Extents>
 template <class... SliceSpecifiers>
-MDSPAN_INLINE_FUNCTION
-constexpr auto
-layout_left::mapping<Extents>::submdspan_mapping_impl(SliceSpecifiers... slices) const {
+MDSPAN_INLINE_FUNCTION constexpr auto
+layout_left::mapping<Extents>::submdspan_mapping_impl(
+    SliceSpecifiers... slices) const {
 
   // compute sub extents
   using src_ext_t = Extents;
   auto dst_ext = submdspan_extents(extents(), slices...);
   using dst_ext_t = decltype(dst_ext);
 
   // figure out sub layout type
-  constexpr bool preserve_layout = detail::preserve_layout_left_mapping<
-      decltype(std::make_index_sequence<src_ext_t::rank()>()), dst_ext_t::rank(),
-      SliceSpecifiers...>::value;
-  using dst_layout_t =
-      std::conditional_t<preserve_layout, layout_left, layout_stride>;
+  using deduce_layout = detail::deduce_layout_left_submapping<
+      typename dst_ext_t::index_type, dst_ext_t::rank(),
+      decltype(std::make_index_sequence<src_ext_t::rank()>()),
+      SliceSpecifiers...>;
+
+  using dst_layout_t = std::conditional_t<
+      deduce_layout::layout_left_value, layout_left,
+      std::conditional_t<
+          deduce_layout::layout_left_padded_value,
+          MDSPAN_IMPL_PROPOSED_NAMESPACE::layout_left_padded<dynamic_extent>,
```

Comment:
I might be missing something, but does this not implement the `layout_left_padded` with static padding stride case?  That should be possible, as in the following example.

```c++
auto mapping = layout_left::mapping<extents<int, 10, 10>>{};
auto x_storage = std::array<float, 100>{};
auto x = mdspan{x_storage.data(), mapping};
auto slice = strided_slice{
  .offset=0,
  .extent=integral_constant<int, 5>{},
  .stride=integral_constant<int, 1>{}};
// Should be `layout_left_padded<10>::mapping<extents<int, 5, 5>>`
auto x_sub = submdspan(x, strided_slice, full_extent);
```

Does the wording not permit this, or is that just future work?  Totally OK if so.

Reply:
The standard actually does not allow this. I.e. according to the standard you get layout_stride. I am not sure I want to change that, i.e. explicitly treating the strided_slice with integral_constant 1 for stride like a pair range. I mean we could if we really wanted to, not sure it wins us much. And it might be weird in generic code to get layout_stride vs layout_left or so back depending on an integral constant. In all other cases the layout_flavor you get really just depends on the slice flavors you have, not the particular type of slice.

That said: this PR always returns dynamic_extent for the padding value when it returns padded layouts. I am fixing that in the follow up which actually introduces submdspan for padded layouts. That anyway requires a bunch of more extensive new tests.

Reply:
Yes, I am doing it as part of the padded layouts submdspan. Should be done later today or tomorrow.